docs/tutorials/pytorch.ipynb

```diff
@@ -0,0 +1,507 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d235772d-475e-42e7-bc7c-f50729ee0e22",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def train(dataloader):\n",
+    "    model.train()\n",
+    "    total_loss = 0\n",
+    "    for batch in dataloader:\n",
+    "        x = batch['image'].to(device)\n",
+    "        y = batch['label'].to(device)\n",
+    "\n",
+    "        # Forward pass\n",
+    "        y_hat = model(x)\n",
+    "        loss = loss_fn(y_hat, y)\n",
+    "        total_loss += loss.item()\n",
+    "\n",
+    "        # Backward pass\n",
+    "        loss.backward()\n",
+    "        optimizer.step()\n",
+    "        optimizer.zero_grad()\n",
+    "\n",
+    "    print(f'Loss: {total_loss:.2f}')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1fd82312-cd17-4886-bcb6-8e42633e5009",
+   "metadata": {},
+   "source": [
+    "## Evaluation\n",
+    "\n",
+    "Once the model is trained, we need to evaluate its performance on unseen data. To do this, we set the model to evaluation mode, then iterate over all mini-batches in the dataset. Note that we also disable the computation of gradients, since we do not need to backpropagate them. Finally, we compute the number of correctly classified images."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3bddce3b-ed2f-4a5c-b3c6-2f1a3a51c2d9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def evaluate(dataloader):\n",
+    "    model.eval()\n",
+    "    correct = 0\n",
+    "    with torch.no_grad():\n",
+    "        for batch in dataloader:\n",
+    "            x = batch['image'].to(device)\n",
+    "            y = batch['label'].to(device)\n",
+    "\n",
+    "            # Forward pass\n",
+    "            y_hat = model(x)\n",
+    "            correct += (y_hat.argmax(1) == y).type(torch.float).sum().item()\n",
+    "\n",
+    "    correct /= len(dataloader.dataset)\n",
+    "    print(f'Accuracy: {correct:.0%}')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f62a54e5-897d-476c-8d84-381993dbabbd",
+   "metadata": {},
+   "source": [
+    "## Putting It All Together\n",
+    "\n",
+    "In machine learning, we typically iterate over our datasets multiple times. Each full pass through the dataset is called an *epoch*. The following hyperparameter controls the number of epoch for which we train our model, and can be modified to train the model for longer:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "eb5dc7e8-6cb3-4457-83ad-7fa5aef8ea0c",
+   "metadata": {
+    "nbmake": {
+     "mock": {
```

Comment:
The tutorial runs for 100 epochs, but we mock it to only run for a single epoch during testing.

Reply:
This new tutorial only takes 9.69s to test.